src/components/messaging/chatFileMessage.component.tsx

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import {
+  ImageProps,
+  ImageBackgroundProps,
+  ImageBackground,
+  View,
+  Image,
+  ViewProps,
+} from 'react-native';
+import { FileType } from '@src/core/model';
+import {
+  StyleType,
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+import { Text } from '@src/components/common';
+import {
+  ChatMessage,
+  ChatMessageAlignment,
+  ComponentProps,
+} from './chatMessage.component';
+import { PlayCircleIcon } from '@src/assets/icons';
+
+export enum ChatFileMessageAppearance {
+  full = 'full',
+  preview = 'preview',
+}
+
+interface FileComponentProps {
+  appearance: ChatFileMessageAppearance;
+}
+
+export type ChatFileMessageProps = ComponentProps & FileComponentProps & ThemedComponentProps & ViewProps;
+
+class ChatFileMessageComponent extends React.Component<ChatFileMessageProps> {
+
+  static defaultProps: Partial<ChatFileMessageProps> = {
+    alignment: ChatMessageAlignment.left,
+  };
+
+  private getAlignmentContainerStyle = (): StyleType => {
+    const { themedStyle, alignment } = this.props;
+
+    if (alignment === ChatMessageAlignment.left) {
+      return themedStyle.containerLeft;
+    } else {
+      return themedStyle.containerRight;
+    }
+  };
+
+  private renderFullMessageAppearanceImage = (): React.ReactElement<ImageProps | ImageBackgroundProps> => {
+    const { message, themedStyle } = this.props;
+
+    return message.file.type === FileType.photo ? (
+      <Image
+        style={themedStyle.fullMessageImage}
+        source={{ uri: message.file.preview }}
+      />
+    ) : (
+      <ImageBackground
+        style={themedStyle.fullMessageVideo}
+        imageStyle={themedStyle.fullMessageVideoPreview}
+        source={{ uri: message.file.preview }}>
+        {PlayCircleIcon(themedStyle.fullMessageVideoPreviewIcon)}
+      </ImageBackground>
+    );
+  };
+
+  private renderFullMessageAppearance = (): React.ReactElement<ViewProps> => {
+    const {
+      index,
+      alignment,
+      themedStyle,
+      message,
+    } = this.props;
+
+    return (
+      <ChatMessage
+        index={index}
+        alignment={alignment}
+        message={message}>
+        <Text style={themedStyle.fileNameLabel}>{message.file.name}</Text>
+        {this.renderFullMessageAppearanceImage()}
+      </ChatMessage>
+    );
+  };
+
+  private renderPreviewMessageAppearanceImage = (): React.ReactElement<ImageBackgroundProps | ImageProps> => {
+    const { message, themedStyle } = this.props;
+
+    return message.file.type === FileType.photo ? (
+      <Image
```

Comment:
How about using `FileType` to provide corresponding view?

Reply:
Didn't catch your point. Could you please explain in little bit more details?

Reply:
Done